commit ad3330f70affc40910856c88cf777ed47837a006
Author: Jeff Law <jlaw@ventanamicro.com>
Date:   Tue Apr 1 09:40:12 2025 -0600

    Revert "PR middle-end/119442: expr.cc: Fix vec_duplicate into vector boolean modes"
    
    This reverts commit 70391e3958db791edea4e877636592de47a785e7.

diff --git a/gcc/expr.cc b/gcc/expr.cc
index 2147eedad7b..9f4382d7986 100644
--- a/gcc/expr.cc
+++ b/gcc/expr.cc
@@ -7920,16 +7920,11 @@ store_constructor (tree exp, rtx target, int cleared, poly_int64 size,
 	gcc_assert (eltmode != BLKmode);
 
 	/* Try using vec_duplicate_optab for uniform vectors.  */
-	icode = optab_handler (vec_duplicate_optab, mode);
 	if (!TREE_SIDE_EFFECTS (exp)
 	    && VECTOR_MODE_P (mode)
-	    && icode != CODE_FOR_nothing
-	    /* If the vec_duplicate target pattern does not specify an element
-	       mode check that eltmode is the normal inner mode of the
-	       requested vector mode.  But if the target allows eltmode
-	       explicitly go ahead and use it.  */
-	    && (eltmode == GET_MODE_INNER (mode)
-		|| insn_data[icode].operand[1].mode == eltmode)
+	    && eltmode == GET_MODE_INNER (mode)
+	    && ((icode = optab_handler (vec_duplicate_optab, mode))
+		!= CODE_FOR_nothing)
 	    && (elt = uniform_vector_p (exp))
 	    && !VECTOR_TYPE_P (TREE_TYPE (elt)))
 	  {
diff --git a/gcc/testsuite/gcc.target/aarch64/vls_sve_vec_dup_1.c b/gcc/testsuite/gcc.target/aarch64/vls_sve_vec_dup_1.c
deleted file mode 100644
index ada0d4fc0a4..00000000000
--- a/gcc/testsuite/gcc.target/aarch64/vls_sve_vec_dup_1.c
+++ /dev/null
@@ -1,15 +0,0 @@
-/* { dg-do compile } */
-/* { dg-options "-O3 -march=armv8.2-a+sve -msve-vector-bits=128" } */
-
-float fasten_main_etot_0;
-void fasten_main() {
-  for (int l = 0; l < 2;) {
-    int phphb_nz;
-    for (; l < 32; l++) {
-      float dslv_e = l && phphb_nz;
-      fasten_main_etot_0 += dslv_e;
-    }
-  }
-}
-
-/* { dg-final { scan-assembler-not {bfi\tw\[0-9\]+} } } */
